paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -162,6 +212,40 @@ internal class CustomerSheetViewModel(
                 loadCustomerSheetState()
             }
         }
+
+        viewModelScope.launch {
+            selectPaymentMethodState.collectLatest { selectPaymentMethodState ->
+                updateViewState<CustomerSheetViewState.SelectPaymentMethod> {
+                    selectPaymentMethodState
+                }
+            }
+        }
+
+        viewModelScope.launch {
+            viewState.collectLatest { viewState ->
+                if (
+                    viewState is CustomerSheetViewState.SelectPaymentMethod &&
+                    !customerState.value.canShowSavedPaymentMethods
```

Comment:
should `customerState` also be part of the `collect` here? ie if `customerState` changed but `viewState` didn't, should we end up in this update flow?

Reply:
I'm not sure if the update matched what I meant -- it looks like we should be updating this value when either viewState or customerState changes, not when just one of them changes. Is that right? If so, we should probably combine the two flows and then collect on that

Reply:
Ah my bad. We actually only actually only care if the `ViewState` is `SelectPaymentMethod` at the time of the update. I updated the view state to check if we have saved payment options instead so we only have to observe the view state.